src/sqlfluff/core/parser/segments/base.py

```diff
@@ -683,19 +683,25 @@ def select_children(
                 buff.append(seg)
         return buff
 
-    def recursive_crawl(self, *seg_type):
+    def recursive_crawl(self, *seg_type, recurse_into=True):
```

Comment:
New, optional parameter lets you avoid descending inside matching segments.

Reply:
Interesting! So does that mean you're not looking for a SELECT that is a child of another SELECT ? (I imagine that would only be the case for a subquery?)

Reply:
We are definitely interested in nested queries, but not initially. The overall design is:
* Find all the top-level queries, i.e. the "main" query and all the CTEs.
* Starting from the main query, follow all the `SELECT *` dependencies, which can take us in three "directions":
  * To a CTE
  * To a nested query
  * To an external table

We don't need to find the nested stuff up front because we're not trying to explore the full query. We're just searching until we hit one of:
* External table
* Query that does **not** use `*`